### PR TITLE
115: fix installation method

### PR DESCRIPTION
**Pull request by Loyalsoldier**
After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

---

### CODE REVIEW

File: Casks/1/115.rb [hunk 1]
```diff
@@ -18,13 +18,27 @@
...
+  container type: :naked
 
-  app "115生活.app"
+  preflight do
+    system_command "/usr/bin/hdiutil",
+                   args: ["attach", "-nobrowse", "#{staged_path}/115pc_v#{version}.dmg"]
+    system_command "/Volumes/115生活/115生活.app/Contents/MacOS/115setup"
+    system_command "/usr/bin/hdiutil",
+                   args: ["detach", "/Volumes/115生活"]
+  end
```
Comment (bevanjkay) [inline Casks/1/115.rb:29]: Preflight blocks should be used very sparingly. 

Can we not run the setup script using the `script` stanza?

Comment (Loyalsoldier) [reply Casks/1/115.rb:29]: The `115setup` script is a binary, I don't know what it does.

Comment (bevanjkay) [reply Casks/1/115.rb:29]: Can we try running it with the `script` stanza instead then?
Note, that if the app is installed this way the install should not require user-intervention.

Comment (Loyalsoldier) [reply Casks/1/115.rb:29]: Can `installer script` stanza mount / unmount dmg?

Comment (bevanjkay) [reply Casks/1/115.rb:29]: It shouldn't need to. You can use `container: :nested` 
Example - https://github.com/Homebrew/homebrew-cask/blob/aa461148bbb5119af26b82cccf5003e2b4e50d95/Casks/b/blocs.rb#L17-L19

Comment (Loyalsoldier) [reply Casks/1/115.rb]: I tried to use `container: :nested` but the setup script uses many files in the dmg, so for convenience I mount / unmount the dmg in `installer script` stanza.

Comment (Loyalsoldier) [reply Casks/1/115.rb]: The setup script `115setup` uses many files in the dmg file. I tried to use `container: :nested` but it cannot extract directory. For convenience I mount / unmount the dmg in `installer script` stanza.

Comment (bevanjkay) [reply Casks/1/115.rb]: Sorry, I should have realised the container stanza isn't required at all. I have pushed an update that simplifies this significantly.

File: Casks/1/115.rb [hunk 2]
```diff
@@ -19,12 +19,19 @@
...
-  app "115生活.app"
+  installer script: "#{staged_path}/115生活.app/Contents/MacOS/115setup"
 
   uninstall quit: "org.115pc.115Desktop"
+  uninstall delete: [
+    "/Applications/115生活.app",
+    "/Applications/115生活.app/Contents/app/115DocViewer.app",
+    "/Applications/115生活.app/Contents/app/115Music.app",
+    "/Applications/115生活.app/Contents/app/115Photo.app",
+    "/Applications/115生活.app/Contents/app/115Player.app",
```
[the remark this answers is not in the document]
Comment (Loyalsoldier) [reply Casks/1/115.rb:30]: Then CI reports errors.

**Review by bevanjkay — APPROVED** [review]

**Review by krehel — APPROVED** [review]

**Pull request merged**